File: src/2025-02-22-00-51-50--a-deepseek-r1-test-220-10-1--2-1-huoshan/go/exercises/practice/wordy/wordy.go
package wordy

import (
	"strconv"
	"strings"
)

func Answer(question string) (int, bool) {
	// Check question format
	if !strings.HasPrefix(question, "What is ") || !strings.HasSuffix(question, "?") {
		return 0, false
	}

	// Extract body and handle empty case
	body := strings.TrimSpace(strings.TrimSuffix(strings.TrimPrefix(question, "What is "), "?"))
	if body == "" {
		return 0, false
	}

	// Normalize operations to single tokens
	body = strings.ReplaceAll(body, "multiplied by", "multiply")
	body = strings.ReplaceAll(body, "divided by", "divide")

	tokens := strings.Split(body, " ")

	// Validate token count and pattern
	if len(tokens) != 1 && len(tokens)%2 == 0 {
		return 0, false
	}

	var result int
	for i, token := range tokens {
		if i%2 == 0 { // Number position
			num, err := strconv.Atoi(token)
			if err != nil {
				return 0, false
			}

			if i == 0 {
				result = num
			} else {
				op := tokens[i-1]
				switch op {
				case "plus":
					result += num
				case "minus":
					result -= num
				case "multiply":
					result *= num
				case "divide":
					if num == 0 {
						return 0, false // Division by zero
					}
					result /= num
				default:
					return 0, false // Invalid operation
				}
			}
		} else { // Operation position
			if token != "plus" && token != "minus" && token != "multiply" && token != "divide" {
				return 0, false
			}
		}
	}

	return result, true
}

File: src/2025-02-22-00-51-50--a-deepseek-r1-test-220-10-1--2-1-huoshan/python/exercises/practice/wordy/wordy.py
import re

def answer(question):
    # Validate question structure and extract expression
    question = question.strip()
    if not question.startswith('What is ') or not question.endswith('?'):
        raise ValueError("syntax error")
    
    expr = question[8:-1].strip()
    if not expr:
        raise ValueError("syntax error")
    
    # Normalize operations to single words
    expr = expr.replace('multiplied by', 'multiply').replace('divided by', 'divide')
    tokens = expr.split()
    
    # Validate we have at least a number
    if not tokens:
        raise ValueError("syntax error")
    
    # Parse first number
    try:
        result = int(tokens[0])
    except ValueError:
        raise ValueError("syntax error")
    
    # Process subsequent operations
    i = 1
    while i < len(tokens):
        # Check for operation-number pairs
        if i + 1 >= len(tokens):
            raise ValueError("syntax error")
        
        op = tokens[i]
        number_str = tokens[i+1]
        
        # Validate operation (check if it's a number first)
        try:
            int(op)
            raise ValueError("syntax error")
        except ValueError:
            if op not in ('plus', 'minus', 'multiply', 'divide'):
                raise ValueError("unknown operation")
        
        # Parse number
        try:
            num = int(number_str)
        except ValueError:
            raise ValueError("syntax error")
        
        # Perform calculation
        if op == 'plus':
            result += num
        elif op == 'minus':
            result -= num
        elif op == 'multiply':
            result *= num
        elif op == 'divide':
            if num == 0:
                raise ValueError("syntax error")
            result //= num
        
        i += 2
    
    return result
